node/src/components/linear_chain_sync.rs

```diff
@@ -542,31 +395,32 @@ where
                     self.block_downloaded(rng, effect_builder, block.header())
                 }
             },
-            Event::DeploysFound(block_header) => {
-                let block_height = block_header.height();
-                trace!(%block_height, "deploys for linear chain block found.");
-                // Reset used peers so we can download next block with the full set.
-                self.reset_peers(rng);
-                // Execute block
-                let finalized_block: FinalizedBlock = (*block_header).into();
-                effect_builder.execute_block(finalized_block).ignore()
-            }
-            Event::DeploysNotFound(block_header) => match self.random_peer() {
-                None => {
+            Event::GetDeploysResult(fetch_result) => match fetch_result {
+                event::DeploysResult::Found(block_header) => {
                     let block_hash = block_header.hash();
-                    error!(%block_hash, "could not download deploys from linear chain block.");
-                    panic!("Failed to download linear chain deploys.")
+                    trace!(%block_hash, "deploys for linear chain block found");
+                    // Reset used peers so we can download next block with the full set.
+                    self.peers.reset_peers(rng);
+                    // Execute block
+                    let finalized_block: FinalizedBlock = (*block_header).into();
+                    effect_builder.execute_block(finalized_block).ignore()
                 }
-                Some(peer) => {
-                    let block_hash = (*block_header).hash();
-                    trace!(%block_hash, next_peer=%peer,
-                        "failed to download deploys from a peer. Trying next one");
-                    fetch_block_deploys(effect_builder, peer, *block_header)
+                event::DeploysResult::NotFound(block_header, peer) => {
+                    let block_hash = block_header.hash();
+                    trace!(%block_hash, %peer, "deploy for linear chain block not found. Trying next peer");
+                    match self.peers.random_peer() {
+                        None => {
+                            let block_hash = block_header.hash();
```

Comment:
Do we need this line?  We already have the `block_hash` here I think.

Reply:
Right, we shouldn't need it. Good catch.

Reply:
Done in [`a9dcbf7` (#783)](https://github.com/CasperLabs/casper-node/pull/783/commits/a9dcbf708b8cfd18f5ca74e13d958615ac5dc9f1)